xdsl/dialects/emitc.py

```diff
@@ -81,6 +115,60 @@ def print_parameters(self, printer: Printer) -> None:
             printer.print_attribute(self.element_type)
 
 
+def is_supported_integer_type(type_attr: Attribute) -> bool:
+    """Check if an IntegerType is supported by EmitC."""
+    if isinstance(type_attr, IntegerType):
+        return type_attr.width.data in [1, 8, 16, 32, 64]
+    return False
```

Comment:
So we actually have a mechanism for this sort of thing, and that's an AttributeConstraint, which can be seen as a function from Attribute to bool, except that in the case of False it also has an error message. Could you rewrite these as constraints, and use them in the field definition directly? It might be worth looking at `PositiveIntAttr` in `test_attribute_definition.py`. You're welcome to translate everything in one go or just migrate this and ping me on Zulip so we can iterate.

Reply:
oh nice, sure will do! Thanks a lot for your help :)

Reply:
Actually I think this would also work as `I1 | I8 | I16` and so on